### DISCUSSION TITLE
PIP_INDEX_URL behavior

### DISCUSSION THREAD

**Discussion by ericfrederich** [python-poetry/poetry#4078] [Q&A]
Is there a way to get the behavior of PIP_INDEX_URL using only environment variables?

For instance, on an internal CI build on a machine with no internet access with pip I can access our internal PyPI mirror/cache by doing the following:
```bash
export PIP_INDEX_URL=https://my_user:my_password@some-internal-server.example.com"
pip install .
```

Trying to do the same with a poetry based project I run into issues.  I can set the username and password like the following, but how do I tell poetry to use the internal server?
```bash
export POETRY_HTTP_BASIC_INTERNAL_USERNAME=my_user
export POETRY_HTTP_BASIC_INTERNAL_PASSWORD=my_password
poetry install
```

As described in [Install dependencies from a private repository](https://python-poetry.org/docs/repositories/#install-dependencies-from-a-private-repository).
1. it's stored in `pyproject.toml` not in `~/.config/pypoetry/config.toml`
2. it's an array value for `[[tool.poetry.source]]`` with `name` and `url` keys

Does this mean it's not something that can be configured with environment variables alone?

**Comment by Agalin** [comment]
Check [this](https://github.com/python-poetry/poetry/issues/1632) issue. Currently it's easiest to use `poetry export` but it won't work when you need to add a package. For that you can switch to legacy installer mode in Poetry - it uses pip under the hood so `PIP_INDEX_URL` should work as expected.
